pkg/controller/registry/resolver/cache.go

```diff
@@ -354,10 +368,14 @@ func (s SortableSnapshots) Less(i, j int) bool {
 		return false
 	}
 
-	// the rest are sorted first in namespace preference order, then by name
+	// the rest are sorted first on priority, namespace and then by name
```

Comment:
Not really objecting this, but based on the previous comments, I thought we want to prioritize same ns, then priority, then ns lexicographic,...

I am not objecting the idea to have priority going before ns, but it does not seem like what we agreed on. unless @ecordell changed his mind?

Reply:
resolved. priority goes first because it gives user more control